crates/core/component/dex/src/component/router/fill_route.rs

```diff
@@ -48,8 +48,10 @@ pub trait FillRoute: StateWrite + Sized {
         input: Value,
         hops: &[asset::Id],
         spill_price: Option<U128x128>,
-    ) -> Result<(Value, Value)> {
-        fill_route_inner(self, input, hops, spill_price, true).await
+    ) -> Result<SwapExecution> {
+        Ok(SwapExecution::new(
+            fill_route_inner(self, input, hops, spill_price, true).await?,
```

Comment:
Shouldn't `fill_route_inner` already have computed all of the fields of the `SwapExecution`? In that case, could we just have `fill_route_inner` return a `SwapExecution` directly?